services/src/user/user-repository.integration.test.ts

```
@@ -46,4 +46,20 @@ describe('User Repository', () => {
 			await expect(getUserById(-1)).rejects.toThrow()
 		})
 	})
+
+	describe('deleteUser', () => {
+		it('should delete a user by ID', async () => {
+			await createUser(userCreationObject)
+
+			const createdUser = await db.selectFrom('users').select('id').executeTakeFirstOrThrow()
+
+			const result = await deleteUserById(createdUser.id)
+			expect(result.numDeletedRows).toBe(1n)
+		})
+
+		it('should return 0 when the user not exists', async () => {
```

I think returning 0 here is not the best way to go. I think that generally, the function should not return anything at all, since there is no value returning the user id (as it is known already). And to signify success: When there is an error, an error will be thrown, otherwise we assume success.

How we want to handle non-existing keys is a different question. imo it should be an error, but I can also be convinced that since it's an idempotent operation, we just don't do anything. The point of the function is that after calling it, there is no user with that ID. If there isn't one in the first place that is still true.

---

Agree, let's **not** return a 0 here. Let's throw an error here, of course we can make it idempotent, but I do not see the necessity for that.

---

No I also don't see the necessity for idempotency, but I mean that from an SQL perspective it would be/there is an argument why it would be okay. I agree with the error. After all deleting something that doesn't exist should usually not happen